flink-core/src/main/java/org/apache/flink/api/java/typeutils/runtime/DataInputViewStream.java

```diff
@@ -62,7 +62,7 @@ public long skip(long n) throws IOException {
 
 			counter -= skippedBytes;
 		}
-		return n - counter - inputView.skipBytes((int) counter);
+		return n - counter + inputView.skipBytes((int) counter);
```

Comment:
A little suggestion: we can change the variable name `counter` to some like `toSkipRemaining` or other you think better by the way to make it more explicit.

Reply:
I will apply the change.